<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Document</title>
</head>

<body>
  <script>
    let time = 0
    let timeIncre = setInterval(() => {
      time++
    }, 1000)

    let p = new Promise((resolve, reject) => {
      setTimeout(() => {
        resolve("第一层")
      }, 1000)
    })
    p.then(
      value => {
        console.log(value)
        console.log("第" + time + "秒")
        return Promise.reject("error")
      },
      reason => { }
    ).then(
      value => {
        console.log(value)
      },
      reason => {
        console.log(reason)
      }
    ).then(value => {
      console.log(value, "这里能显示 因为链式调用没有断")
    }).catch(reason => {
      console.log("这里不会显示 因为前面已经有错误处理了")
    })

    let p_2 = new Promise((resolve, reject) => {
      setTimeout(() => {
        reject("第二层")
      }, 5000)
    })
    p_2.catch(reason => {
      console.log("第" + time + "秒")
      console.log(reason)
      return 521
    }).then(
      value => {
        console.log("我会输出 因为catch不能阻断", value)
      },
      reason => {
        console.log(reason)
      }
    )
  </script>
</body>

</html>